test/tabs/test-firefox-tabs.js

```diff
@@ -1261,6 +1261,40 @@ exports["test ready event after window.open"] = function (assert, done) {
   });
 }
 
+// related to bug #989288
+// https://bugzilla.mozilla.org/show_bug.cgi?id=989288
+exports["test tabs events after window.open"] = function(assert, done) {
+  // ensure popups open in a new windows and disable popup blocker
+  setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+  setPref(DISABLE_POPUP_PREF, false);
+
+  let counter = 0;
+  tabs.on('open', function onOpen(tab) {
+    assert.pass("tab open has occured");
+    counter++;
+    tabs.removeListener('open', onOpen);
+  });
+  tabs.on("ready", function onReady(tab) {
+    assert.pass("tab ready has occured");
+    counter++;
+    tab.close();
+    tabs.removeListener('ready', onReady);
+  });
+  tabs.on("close", function onClose(tab) {
+    assert.pass("tab close has occured");
+    counter++;
+    tabs.removeListener('close', onClose);
+    assert.equal(counter, 3, "All expected tab events have occured");
+    done();
+  });
+
+  // open window to trigger observers
+  tabs.activeTab.attach({
```

Comment:
I would also tests the scenario where we open multiple dom windows, and check that they're behaving as expected. You can either extend this test or create a new ones.